scipy/stats/tests/test_qmc.py

```diff
@@ -534,23 +535,39 @@ def test_sample(self, *args):
         pytest.skip("Not applicable: the value of reference sample is"
                     " implementation dependent.")
 
+    @pytest.mark.parametrize("orthogonal", [False, True])
+    @pytest.mark.parametrize("centered", [False, True])
     @pytest.mark.parametrize("optimization", [None, "random-CD"])
-    def test_sample_stratified(self, optimization):
-        d, n = 4, 20
+    def test_sample_stratified(self, optimization, centered,
+                               orthogonal):
+        d, n = 4, 25
         expected1d = (np.arange(n) + 0.5) / n
         expected = np.broadcast_to(expected1d, (d, n)).T
 
-        engine = self.engine(d=d, scramble=False, centered=True,
+        engine = self.engine(d=d, scramble=False, centered=centered,
+                             orthogonal=orthogonal,
                              optimization=optimization)
         sample = engine.random(n=n)
+        assert sample.shape == (n, d)
+
         sorted_sample = np.sort(sample, axis=0)
 
-        assert_equal(sorted_sample, expected)
         assert np.any(sample != expected)
-
         assert_allclose(sorted_sample, expected, atol=0.5 / n)
         assert np.any(sample - expected > 0.5 / n)
 
+    @pytest.mark.parametrize("centered", [False, True])
+    def test_orthogonal_array(self, centered):
+        p = 5
+
+        engine = self.engine(d=3, scramble=False, centered=centered,
```

Comment:
If the speed is not a concern, I would go for `d=p+1` just in case.

Reply:
I am testing the limit (failing at `d=p+2`) in the error section if this is your concern.

Reply:
Still I updated this test.

Reply:
> Still I updated this test.

Thanks!

> I am testing the limit (failing at `d=p+2`) in the error section (`test_warning`) if this is your concern.

My concern was that the later dimensions (>3) of the OA are always not tested for orthogonality, which IMO undermines the effectiveness of this test for orthogonality.

Reply:
Make sense, good then that I updated the test. Thanks.

Reply:
@kstoneriv3 is the rest ok for you? If yes, can you approve?